internal/controller/preservation/preservation.go

```diff
@@ -187,25 +124,36 @@ func Cleanup(
 ) (completedCleanup bool, err error) {
 	completedCleanup = true
 
-	// Define the label selector for filtering ConfigMaps and Secrets.
-	labelSelector, err := buildRestoreLabelSelector(clusterInstance.Spec.Reinstall.PreservationMode)
-	if err != nil {
-		return false, err
+	namespace := clusterInstance.Namespace
+	modes := []v1alpha1.PreservationMode{
+		clusterInstance.Spec.Reinstall.PreservationMode,
+		PreservationModeInternal,
 	}
 
-	if labelSelector == nil {
-		return true, nil
-	}
+	var data resources
+	for _, mode := range modes {
 
-	listOptions := &client.ListOptions{
-		Namespace:     clusterInstance.Namespace,
-		LabelSelector: labelSelector,
-	}
+		// Define the label selector for filtering ConfigMaps and Secrets.
+		labelSelector, err := buildRestoreLabelSelector(mode)
+		if err != nil {
+			return false, err
+		}
+		if labelSelector == nil {
+			continue
+		}
+
+		listOptions := &client.ListOptions{
+			Namespace:     namespace,
+			LabelSelector: labelSelector,
+		}
 
-	// Retrieve data for restoration
-	data, err := resourcesForPreservation(ctx, c, log, listOptions)
-	if err != nil {
-		return false, err
+		// Retrieve data for restoration
+		dAtA, err := resourcesForPreservation(ctx, c, log, listOptions)
```

Comment:
This is very strange capitalization.

Reply:
I've used `dAtA` to convey that data objects are retrieved and appended to the outer / main `data` variable.

Reply:
Changing capitalization doesn't convey anything.

Maybe `resourcesForMode` would be a better way to actually give a reader more information about this.

Reply:
I'll change `dAtA` to something more aesthetically pleasing :)